Sources/Components/Feed/InAppFeedViewModel.swift

```diff
@@ -0,0 +1,312 @@
+        @Published public var topButtonActions: [Knock.FeedTopActionButtonType]? /// Actions available at the top of the feed interface.
+
+        @Published internal var brandingRequired: Bool = true /// Controls whether to show the Knock icon on the feed interface.
+        @Published var showRefreshIndicator: Bool = false
+        @Published public var currentFilter: InAppFeedFilter { /// The currently selected filter for displaying feed items.
+            didSet {
+                filterDidChange()
+            }
+        }
+        
+        public var feedClientOptions: Knock.FeedClientOptions /// Configuration options for feed.
+        public let didTapFeedItemButtonPublisher = PassthroughSubject<String, Never>() /// Publisher for feed item button tap events.
+        public let didTapFeedItemRowPublisher = PassthroughSubject<Knock.FeedItem, Never>() /// Publisher for feed item row tap events.
+        
+        // MARK: Initialization
+        
+        public init(
+            feedClientOptions: Knock.FeedClientOptions = .init(),
+            currentTenantId: String? = nil,
+            currentFilter: InAppFeedFilter? = nil,
+            filterOptions: [InAppFeedFilter]? = nil,
+            topButtonActions: [Knock.FeedTopActionButtonType]? = [.markAllAsRead(), .archiveRead()]
+        ) {
+            self.feedClientOptions = feedClientOptions
+            self.currentTenantId = currentTenantId ?? feedClientOptions.tenant
+            self.filterOptions = filterOptions ?? [.init(scope: .all), .init(scope: .unread), .init(scope: .archived)]
+            self.currentFilter = currentFilter ?? filterOptions?.first ?? .init(scope: .all)
+            self.topButtonActions = topButtonActions
+            
+            // Assign initial settings to _feedClientOptions to be used in API calls
+            self.feedClientOptions.status = self.currentFilter.scope
+            self.feedClientOptions.tenant = self.currentTenantId
+        }
+        
+        /// Sets up feed manager connection and subscribes to new message events.
+        public func connectFeedAndObserveNewMessages() async {
+            Knock.shared.feedManager?.connectToFeed()
+            Knock.shared.feedManager?.on(eventName: "new-message") { [weak self] _ in
+                guard let self = self else { return }
+                feedClientOptions.before = self.feed.pageInfo.before
+                Knock.shared.feedManager?.getUserFeedContent(options: feedClientOptions) { result in
+                    DispatchQueue.main.async {
+                        switch result {
+                        case .success(let feed):
+                            self.mergeFeedsForNewMessageReceived(feed: feed)
+                        case .failure(let error):
+                            self.handleFeedError(error)
+                        }
+                    }
+                }
+            }
+            await refreshFeed(showRefreshIndicator: true)
```

Comment:
Should we be doing these two methods here? Feels like maybe they should be separate

Reply:
@cjbell for this are you recommending that we remove the refreshFeed and the branding logic from the connectFeedAndObserveNewMessages method?